src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -502,6 +507,32 @@ public void replaceVariablesInDirectoryName(String startPath, String targetStrin
         }
     }
 
+    /**
+     * This replace all occurrences of the target String with the replacement String within a source file of a given directory (recursive!)
+     *
+     * @param startPath         the path where the file is located
+     * @param targetString      the string that should be replaced
+     * @param replacementString the string that should be used to replace the target
+     * @throws IOException if an issue occurs on file access for the replacement of the variables.
+     */
+    public void replaceVariablesInFileName(String startPath, String targetString, String replacementString) throws IOException {
+        log.debug("Replacing {} with {} in directory {}", targetString, replacementString, startPath);
+        File directory = new File(startPath);
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new RuntimeException("Files in the directory " + startPath + " should be replaced but it does not exist.");
```

Comment:
Why not just throw an `IOException`?

Reply:
because all errors in the FileService.java are thrown as a RuntimeException. This might be required.